### PR TITLE
fix not exporting tslint with project rules

### LINKED ISSUE #30: Tslint with exportedConfig and lynt  have different output
Using tslint with the generated configuration with `--exportConfig` doesn't show all errors `lynt` shows

```
$ npx lynt --exportConfig

✔ tslint.json generated

$ npx lynt

/home/daniel/Documents/projects/ledd/scroll-utility/test/automate/index.ts
  43:15  Invalid 'await' of a non-Promise value.  await-promise

✖ 1 lynt error

$ npx tslint -p .
{ no output }
$ 
```

My lynt config in package.json:
```
  "lynt": {
    "typescript": true,
    "react": true,
    "rules": {
      "no-console": "on"
    }
  },
```

### PR DESCRIPTION
**Pull request by danielpa9708**
closes #30

### CODE REVIEW

File: packages/lynt/src/cli.ts [hunk 1]
```diff
@@ -67,6 +67,10 @@ if (configResults) {
   Object.assign(options, configResults.config)
 }
 
+if (options.typescript && !options.project && paths.length === 0) {
```
Comment (saadq) [inline packages/lynt/src/cli.ts]: I think this should be called `filePaths` instead of `paths`.

**Comment by danielpa9708** [comment]
Should be good now, forgot to compile it :|

**Comment by danielpa9708** [comment]
I think we should put the exportConfig part in the index.ts file, and put the defaults there too, but I think it will take a little bit more of code and maybe you are more comfortable doing it, I could go further and make this if you are ok with this

**Comment by saadq** [comment]
Hmm, the only problem I have with that is that in my opinion I think the Node API should be a simple input/output to process.stdin/process.stdout. It shouldn't do side effects like creating files that `exportConfig` would do, which I think should be exclusively part of the CLI.

I think for now, we can just have the `if` checks in both places and maybe look into improving it later. 

And appreciate your patience with me about this! haha

**Comment by danielpa9708** [comment]
Yeah, the if could be in both part, but mean, the export config part is only available through the cli right now, shouldn't it be in the api too ? I don't want it to generate the tslint/eslint config every time is ran, sorry if I wasn't clear

**Comment by saadq** [comment]
I was thinking that `exportConfig` should only be part of the CLI, not the API. The fact that you're exporting the config generally means that it is for some other tool, not Lynt itself. When using Lynt itself (basically the API), there isn't really any use in exporting the config.

**Comment by danielpa9708** [comment]
I always picture the cli as a way of using the api, but ok, it really doesn't make sense using it for exporting the config, the two ifs looks good.

**Comment by saadq** [comment]
> I always picture the cli as a way of using the api

I definitely agree with that (and that is how Lynt currently works). However, I don't think that means that the CLI can't have extra features that don't belong in the API. In my view, the API itself was just meant to lint files with the options you pass it. So as you mentioned, it probably doesn't make sense to do actions like exporting the config from the API itself.

Thanks again for your help @danielpa9708, it's very much appreciated!

**Pull request merged**